### PR TITLE
S2sport

### PR DESCRIPTION
**Pull request by jedwards4b**
Changes to allow relocation of MOM model by using COMP_ROOT_DIR_OCN instead of assuming a path relative to srcroot.
Also writes namelist file input.nml.mom instead of input.nml, this is to prevent overwriting the input.nml written by fv3.   The fv3_interface will now write input.nml.fv3 and the cmeps buildnml (which always runs last) is used to merge input.nml.mom and input.nml.fv3 into input.nml - if only one of mom and fv3 are in the compset then a link is created to input.nml

### CODE REVIEW

File: cime_config/buildlib [hunk 1]
```diff
@@ -74,25 +77,25 @@ def buildlib(caseroot, libroot, bldroot):
...
-                     os.path.join(comp_root_dir_ocn,"MOM6","src","diagnostics"),
-                     os.path.join(comp_root_dir_ocn,"MOM6","src","equation_of_state"),
-                     os.path.join(comp_root_dir_ocn,"MOM6","src","equation_of_state","TEOS10"),
-                     os.path.join(comp_root_dir_ocn,"MOM6","src","framework"),
-                     os.path.join(comp_root_dir_ocn,"MOM6","src","ice_shelf"),
-                     os.path.join(comp_root_dir_ocn,"MOM6","src","initialization"),
-                     os.path.join(comp_root_dir_ocn,"MOM6","src","ocean_data_assim"),
-                     os.path.join(comp_root_dir_ocn,"MOM6","src","ocean_data_assim", "core"),
-                     os.path.join(comp_root_dir_ocn,"MOM6","src","ocean_data_assim", "geoKdTree"),
+                     os.path.join(comp_root_dir_ocn,"src","config_src",driver),
```
Comment (jedwards4b) [inline cime_config/buildlib:80]: I realize this isn't going to work and will redo this part prior to PR merge.

**Comment by alperaltuntas** [comment]
@jedwards4b Do you plan to update this PR soon, or should I create the tag before this gets merged.

**Comment by jedwards4b** [comment]
@alperaltuntas thanks for the heads up - looking

**Comment by jedwards4b** [comment]
@alperaltuntas I think that you should go ahead and make the tag without this.  Thanks for checking.

**Pull request closed without merging**